Tasks/DotNetCoreInstallerV1/versionFetcher.ts

```diff
@@ -0,0 +1,273 @@
+            proxy: {
+                proxyUrl: proxyUrl,
+                proxyUsername: tl.getVariable("agent.proxyusername"),
+                proxyPassword: tl.getVariable("agent.proxypassword"),
+                proxyBypassHosts: tl.getVariable("agent.proxybypasslist") ? JSON.parse(tl.getVariable("agent.proxybypasslist")) : null
+            }
+        } : {};
+
+        this.httpCallbackClient = new httpClient.HttpClient(tl.getVariable("AZURE_HTTP_USER_AGENT"), null, requestOptions);
+    }
+
+    public async getVersionInfo(version: string, packageType: string, includePreviewVersions: boolean): Promise<VersionInfo> {
+        var requiredVersion: VersionInfo = null;
+        if (this.releasesIndex == null) {
+            await this.setReleasesIndex();
+        }
+
+        let channelInformation: any = this.getVersionChannel(version);
+        if (channelInformation) {
+            requiredVersion = await this.getVersionFromChannel(channelInformation, version, packageType, includePreviewVersions);
+        }
+
+        if (!requiredVersion) {
+            console.log("FallingBackToAdjacentChannels", version);
+            requiredVersion = await this.getVersionFromOtherChannels(version, packageType, includePreviewVersions);
+        }
+
+        if (!requiredVersion) {
+            throw tl.loc("VersionNotFound", version);
+        }
+
+        return requiredVersion;
+    }
+
+    public getDownloadUrl(versionInfo: VersionInfo, packageType: string): string {
+        console.log(tl.loc("GettingDownloadUrl", packageType, versionInfo.version));
+
+        let osSuffixes = this.detectMachineOS();
+        let downloadPackageInfoObject: any = versionInfo.files.find((downloadPackageInfo: any) => {
+            if (downloadPackageInfo.rid.toLowerCase() == osSuffixes[0].toLowerCase()
+                && (osSuffixes[0].split("-")[0] == "win" && downloadPackageInfo.name.endsWith(".zip"))) {
+                return true;
+            }
+
+            return false;
+        });
+
+        if (!!downloadPackageInfoObject && downloadPackageInfoObject.url) {
+            return downloadPackageInfoObject.url;
+        }
+
+        return "";
+    }
+
+    private setReleasesIndex(): Promise<any> {
+        return this.httpCallbackClient.get(DotNetCoreReleasesIndexUrl)
+            .then((response: HttpClientResponse) => {
+                return response.readBody();
+            })
+            .then((body: string) => {
+                this.releasesIndex = JSON.parse(body);
+                return this.releasesIndex;
+            })
+            .catch((ex) => {
+                throw tl.loc("");
+            });
+    }
+
+    private getVersionChannel(version: string): any {
+        let versionParts: any = utils.getVersionParts(version);
+
+        let channelVersion = `${versionParts.majorVersion}.${versionParts.minorVersion}`;
+        if (versionParts.minorVersion == "x") {
```

Comment:
Is the version input validated to only contains number of x for minor/patch versions?

Reply:
Validating now in versionParts class.